framework/Models/Code.py

```diff
@@ -332,6 +332,10 @@ def initialize(self,runInfoDict,inputFiles,initDict=None):
       else:
         self.raiseAMessage('not found pre-executable '+self.executable,'ExceptedError')
 
+    if 'initialize' in dir(self.code):
```

Comment:
We can remove the if check, right? Since we have add this method the code interface base class.

Reply:
we do not strictly require that the CodeInterfaces inherit from the Base Class so I would keep it for now.

Reply:
ok, the changes are good to me.